exla/lib/exla/defn.ex

```diff
@@ -733,6 +733,43 @@ defmodule EXLA.Defn do
     )
   end
 
+  defp to_operator(:take_along_axis, [tensor, indices, axis], _ans, state) do
+    indices_shape = op_shape(indices)
+    indices_rank = tuple_size(indices_shape)
+
+    axes_range = 0..(indices_rank-1)
+
+    index_vector_dim = indices_rank
+    slice_sizes = 1 |> List.duplicate(indices_rank)
+    offset_dims = []
+    collapsed_slice_dims = Enum.to_list(axes_range)
+    start_index_map = Enum.to_list(axes_range)
+
+    iotas =
+      Enum.map(axes_range, fn axis ->
+        shape = EXLA.Op.get_shape(indices)
+        EXLA.Op.iota(state.builder, shape, axis)
+      end)
+
+    new_axis_shape = Tuple.insert_at(indices_shape, indices_rank, 1)
```

Comment:
Probably makes the intent a bit more clear, but whatever you like best!

```suggestion
    new_axis_shape = Tuple.append(indices_shape, 1)
```

Reply:
This is better! I didn't know the function :)

Reply:
@jonatanklosko I've made the suggested changes and I've also added a sanity test with argsort :)

Reply:
Great idea :cowboy_hat_face: